pkgs/applications/science/math/cntk/default.nix

```diff
@@ -33,6 +33,9 @@ in stdenv.mkDerivation rec {
   # Force OpenMPI to use g++ in PATH.
   OMPI_CXX = "g++";
 
+  # Uses some deprecated protobuf functions
+  NIX_CFLAGS_COMPILE = ''-Wno-error=deprecated-declarations'';
```

Comment:
This is the other error after we get past the configure script issue:

```
Source/CNTKv2LibraryDll/tensorboard/tensorboard.pb.cc: In member function 'virtual google::protobuf::uint8* tensorflow::GraphDef::_InternalSerialize(google::protobuf::uint8*, google::protobuf::io::EpsCopyOutputStrea
m*) const':
Source/CNTKv2LibraryDll/tensorboard/tensorboard.pb.cc:5901:21: warning: 'google::protobuf::int32 tensorflow::GraphDef::version() const' is deprecated [-Wdeprecated-declarations]
   if (this->version() != 0) {
                     ^
In file included from Source/CNTKv2LibraryDll/tensorboard/tensorboard.pb.cc:4:0:
Source/CNTKv2LibraryDll/tensorboard/tensorboard.pb.h:8145:39: note: declared here
 inline ::PROTOBUF_NAMESPACE_ID::int32 GraphDef::version() const {
                                       ^~~~~~~~
Source/CNTKv2LibraryDll/tensorboard/tensorboard.pb.cc: In member function 'virtual size_t tensorflow::GraphDef::ByteSizeLong() const':
Source/CNTKv2LibraryDll/tensorboard/tensorboard.pb.cc:5945:21: warning: 'google::protobuf::int32 tensorflow::GraphDef::version() const' is deprecated [-Wdeprecated-declarations]
   if (this->version() != 0) {
                     ^
In file included from Source/CNTKv2LibraryDll/tensorboard/tensorboard.pb.cc:4:0:
Source/CNTKv2LibraryDll/tensorboard/tensorboard.pb.h:8145:39: note: declared here
 inline ::PROTOBUF_NAMESPACE_ID::int32 GraphDef::version() const {
                                       ^~~~~~~~
Source/CNTKv2LibraryDll/tensorboard/tensorboard.pb.cc: In member function 'void tensorflow::GraphDef::MergeFrom(const tensorflow::GraphDef&)':
Source/CNTKv2LibraryDll/tensorboard/tensorboard.pb.cc:5986:20: warning: 'google::protobuf::int32 tensorflow::GraphDef::version() const' is deprecated [-Wdeprecated-declarations]
   if (from.version() != 0) {
                    ^
In file included from Source/CNTKv2LibraryDll/tensorboard/tensorboard.pb.cc:4:0:
Source/CNTKv2LibraryDll/tensorboard/tensorboard.pb.h:8145:39: note: declared here
 inline ::PROTOBUF_NAMESPACE_ID::int32 GraphDef::version() const {
                                       ^~~~~~~~
```

I'm not sure if this is the right way to handle it or not, but it builds at least.

Reply:
```suggestion
  NIX_CFLAGS_COMPILE = "-Wno-error=deprecated-declarations";
```

Reply:
`CFLAGS_COMPILE` does not seem to work, the `deprecated-declarations` end up as errors:

```
Source/CNTKv2LibraryDll/tensorboard/tensorboard.pb.cc: In member function 'void tensorflow::GraphDef::MergeFrom(const tensorflow::GraphDef&)':
Source/CNTKv2LibraryDll/tensorboard/tensorboard.pb.cc:5986:20: error: 'google::protobuf::int32 tensorflow::GraphDef::version() const' is deprecated [-Werror=deprecated-declarations]
   if (from.version() != 0) {
                    ^
```

I'll make the change to double-quotes though.